python/cuml/internals/array.py

```diff
@@ -219,23 +232,21 @@ def __init__(
             mem_type = MemoryType.from_str(mem_type)
         self._mem_type = mem_type
 
-        # Coerce data into an array interface and determine mem_type and owner
-        # if necessary
-        try:
+        if hasattr(data, "__cuda_array_interface__"):
```

Comment:
Did you move from `try: ... except: ...` to `hasattr()` checks because of performance reasons?

Reply:
both potential performance reasons, and the fact that if we decide to go with a similar cumldataframe wrapper asn we do with cumlarray, these `try` can be problematic depending on what the underlying wrapper does